bundles/binding/org.openhab.binding.modbus/src/main/java/org/openhab/binding/modbus/internal/ModbusGenericBindingProvider.java

```diff
@@ -152,14 +153,19 @@ State getItemState() {
             return item.getState();
         }
 
+        public void setItemState(State state) {
+            ((GenericItem) item).setState(state);
+            ;
```

Comment:
Please check the discussion for PR https://github.com/openhab/openhab/pull/4291 (the predecessor of this one). Basically the binding has to cache values somehow to identify changed values. Sami's first idea was to use the ItemRegistry to avoid local caching, but Kai Kreuzer commented in another binding that ItemRegistry should also not be used, so I reworked it to do it the way you see here.

There is another way to do it: Internally store the source value that comes from the modbus server (16 bit or 1 bit), before they are converted to items. If this is the way to go, I can rework the pull request (again) to avoid effects that might happen using the Item for this (although I have this configuration running for many weeks now without problems). But I would really like to clarify this before starting to write the third version of a PR to solve this issue. Sami, John: Would that be ok?